### PR TITLE
rm spaces, const Paramter NO:functionalization

### PR DESCRIPTION
**Pull request by Messikommer**
@rdube ,  @foehnph

### CODE REVIEW

File: obstacle_detection.cpp [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-#include <vector>
-
 #define PI 3.14159265
 
 ros::Publisher obstacle_pub;
 
-
 void scan(const sensor_msgs::PointCloud2& cloud_message) {
-  
+  const float y_grenze_m= 1.5;
```
Comment (rdube) [inline obstacle_detection.cpp:22]: English pls ;-)

File: obstacle_detection.cpp [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+  const float toleranc_m=0.02;
```
Comment (rdube) [inline obstacle_detection.cpp:23]: toleranc(e)

File: obstacle_detection.cpp [hunk 3]
```diff
@@ -42,38 +41,32 @@ void scan(const sensor_msgs::PointCloud2& cloud_message) {
...
         //Store
         if(d_histo.size()==0) {
           std::vector<float> v_dist;
           v_dist.push_back(d);
           v_dist.push_back(1);
           d_histo.push_back(v_dist);
-          
         }
         
         else{
```
Comment (rdube) [inline obstacle_detection.cpp:60]: you can put the brackets on the same line eg. `} else {`

File: obstacle_detection.cpp [hunk 4]
```diff
@@ -88,16 +81,12 @@ void scan(const sensor_msgs::PointCloud2& cloud_message) {
```
Comment (rdube) [inline obstacle_detection.cpp:81]: rm vertical space

File: obstacle_detection.cpp [hunk 5]
```diff
@@ -42,38 +41,32 @@ void scan(const sensor_msgs::PointCloud2& cloud_message) {
...
-             
-             if(a-0.06<= d && d<=a+0.06) {    
+             if(a-toleranc_m<= d && d<=a+toleranc_m) {    
                //Update the average distance of Intervall and the sum
                d_histo[i_2][0]=(d_histo[i_2][1]*d_histo[i_2][0]+d)/(d_histo[i_2][1]+1);
                d_histo[i_2][1]=d_histo[i_2][1]+1;
                included=true;
- 
-               
              }
```
Comment (rdube) [inline obstacle_detection.cpp:71]: rm vertical space

**Review by rdube — COMMENTED** [review]
Good job!

File: obstacle_detection.cpp [hunk 6]
```diff
@@ -1,26 +1,26 @@
...
 #include <pcl/point_types.h>
 #include <pcl/PCLPointCloud2.h>
 #include <pcl/conversions.h>
 #include <pcl_ros/transforms.h>
 
-
-#include <iostream>
-#include <vector>
-
 #define PI 3.14159265
```
Comment (foehnx) [inline obstacle_detection.cpp:17]: `PI` is already defined in math.h

File: obstacle_detection.cpp (hunk 1 above)
Comment (foehnx) [reply obstacle_detection.cpp:22]: use double instead of float. everywhere.

File: obstacle_detection.cpp (hunk 2 above)
Comment (foehnx) [reply obstacle_detection.cpp:23]: again double and spacing.

File: obstacle_detection.cpp [hunk 7]
```diff
@@ -107,95 +96,58 @@ float inter_d=0.0;  //distance most points had
...
-  
-  
   ros::spin();
   return 0;
-}
-
-
-
-
+}
```
Comment (foehnx) [inline obstacle_detection.cpp:153]: one newline after last brace is wanted!

File: obstacle_detection.cpp [hunk 8]
```diff
@@ -1,26 +1,26 @@
...
 ros::Publisher obstacle_pub;
 
-
 void scan(const sensor_msgs::PointCloud2& cloud_message) {
-  
+  const float y_grenze_m= 1.5;
+  const float toleranc_m=0.02;
   
   //converts sensor_msgs::PointCloud2 to pcl::PCLPointCLoud2 and then to 
   pcl::PCLPointCloud2 pcl_pc2_1;
```
Comment (foehnx) [inline obstacle_detection.cpp:26]: Line 29: careful with `new` pointer allocation. better use `std::make_unique`

File: obstacle_detection.cpp [hunk 9]
```diff
@@ -32,7 +32,6 @@ void scan(const sensor_msgs::PointCloud2& cloud_message) {
 //store numbers of points and the distance interval, in which the points can be found
   std::vector< std::vector<float> > d_histo;
   
-  
   //Filter
   
  for(int i_1=0;i_1 <= temp_cloud->size(); i_1++) {
```
Comment (foehnx) [inline obstacle_detection.cpp:37]: with `<=` aren't <ou doing one iteration too much?

File: obstacle_detection.cpp [hunk 10]
```diff
@@ -107,95 +96,58 @@ float inter_d=0.0;  //distance most points had
...
-  
- 
- 
- 
- 
- 
  
  //Second Iteratation to find the points with the distance, which the most points had.
  
  for(int i_4=0;i_4 <= temp_cloud->size(); i_4++) {
```
Comment (foehnx) [inline obstacle_detection.cpp:106]: again, one iteration too much?

File: obstacle_detection.cpp [hunk 11]
```diff
@@ -42,38 +41,32 @@ void scan(const sensor_msgs::PointCloud2& cloud_message) {
    float z=temp_cloud->points[i_1].z;
    
    //check if in Range
-    if (x>0 && -1.5<y && y<1.5 && z<0) {
+    if (x>0 && -y_grenze_m<y && y<y_grenze_m && z<0) {
```
Comment (foehnx) [inline obstacle_detection.cpp:44]: probably use `abs()`

File: obstacle_detection.cpp [hunk 12]
```diff
@@ -42,38 +41,32 @@ void scan(const sensor_msgs::PointCloud2& cloud_message) {
...
         }
         
         else{
           bool included=false;
            for(int i_2=0; i_2<d_histo.size(); i_2++ ) {
              float a=d_histo[i_2][0];
              
-             
-             if(a-0.06<= d && d<=a+0.06) {    
+             if(a-toleranc_m<= d && d<=a+toleranc_m) {
```
Comment (foehnx) [inline obstacle_detection.cpp:65]: use parenthesis to make statement inside if() more readably. probably use `abs()`.

**Review by foehnx — COMMENTED** [review]
Sorry if I commented a lot.

first of all, try to use Eigen for Vector and matrices and not `std::vector` except if this is needed for compatibility to pcl or so.
@rdube  is there any reason for this?
In Eigen you can do stuff like sums and distances much easier.

furthermore, use classes and namespaces defined by `.h` - and `.cc`-files and not just a function.
You will run into compilation/linker errors if you combine your work later on.
If you're just trying out stuff now it's (kind of) ok but I would recommend transcribing this ASAP into a clean objective implementation (node as class).
then you can also do clean publisher/subscriber definitions in the constructor of your class/node.

**Comment by Messikommer** [comment]
@rdube  , @foehnph 
Thanks for checking it out and the helpful input! 
I will most certainly write the code from scratch again.
But first, I want to be sure, that the approach is feasible. If possible, I am going to show the performance of the filter next week.

**Comment by foehnx** [comment]
Hey, perfect. This is absolutely fine.
I will comment when ever I've got time (doesn't matter if you throw it away cause its just a test).
@rdube correct me if I'm wrong in my comments or you guys have other code style conventions.
Next time try even for such simple tests to write it in classes. I'll do a example tomorrow and send it to the team.

File: obstacle_detection.cpp (hunk 9 above)
Comment (rdube) [reply obstacle_detection.cpp:37]: good catch! yes that should be < otherwise you will access outside of the container

**Comment by rdube** [comment]
@foehnph I agree that we should try to use eigen vector or matrices instead of std::vector when applicable. The code style looks similar. We usually use google style at ASL: 

https://google.github.io/styleguide/cppguide.html

Cheers!

**Pull request merged**